src/caja-desktop-window.c

```diff
@@ -174,10 +191,27 @@ caja_desktop_window_new (CajaApplication *application,
     int width_request, height_request;
     int scale;
 
-    scale = gdk_window_get_scale_factor (gdk_screen_get_root_window (screen));
-
-    width_request = WidthOfScreen (gdk_x11_screen_get_xscreen (screen)) / scale;
-    height_request = HeightOfScreen (gdk_x11_screen_get_xscreen (screen)) / scale;
+    GdkDisplay *display = gdk_screen_get_display (screen);
+    if (GDK_IS_X11_DISPLAY (display))
+    {
+        scale = gdk_window_get_scale_factor (gdk_screen_get_root_window (screen));
+        width_request = WidthOfScreen (gdk_x11_screen_get_xscreen (screen)) / scale;
+        height_request = HeightOfScreen (gdk_x11_screen_get_xscreen (screen)) / scale;
+    }
+    else
+    {
+        /*FIXME: There is no primary monitor in wayland itself
+        *compositors can implement this but as this is written
+        *only a few wayland compositors allow setting a primary monitor
+        *and they all do it differently. For now, use the first monitor
+        */
+        GdkRectangle geometry = {0};
+        GdkMonitor *monitor;
+        monitor = gdk_display_get_monitor(display, 0);
+        gdk_monitor_get_geometry(monitor,&geometry);
+        width_request = geometry.width;
+        height_request = geometry.height;
```

Comment:
Shouldn't this use *all* monitors?  in X11, a screen is not a monitor, and nowadays usually there's only one screen no matter the number of monitors.
Ideally I guess there should be one desktop window per monitor to fit any kind of layout, but that's probably a bigger change.

Reply:
The background is in fact drawn (by wayfire itself) across all the monitors. You can only put icons on one of them, and for some reason, recent (pre-wayland) versions of caja seem to have an issue with trying to move icons to a second monitor even in x11.

GTK has deprecated all of the ways of getting a screen's geometry and in fact has removed GtkScreen from GTK4 it seems, thus the deprecations in GTK 3. In wayland I am getting some inconsistant behavior with monitors as well:
if I have two monitors active and call caja with something like `--geometry 4000x4000` it will extend partially to the second monitor and partially overflow the other side of the first. I do NOT know how to fix this, but this is not a common use case I would imagine.

If someone knows a better way I am open to it, but we have  a consensus against adding new deprecated code.

This can wait for another PR if necessary.

Reply:
It's bizarre it wouldn't work on X11, but if it's not this PR I'm OK to fix this later

Reply:
On x11 calling caja oversize such as` --geometry 4000x4000` gives an undecorated fullscreen window exactly filing the primary monitor, it's only on wayland  and then only with a second monitor running that it comes up oversize. Calling caja from a terminal on a secondary monitor in x11 works ONLY if called oversize, and this is with efforts to avoid changing the x11 codepath.

On wayland, caja invoked from terminal appears on the monitor the terminal is on,but can be rendered oversize if and only if a second monitor is active.